internal/dlv/options.go

```diff
@@ -0,0 +1,188 @@
+func WithPort(port int) Option {
+	return With("listen", fmt.Sprintf(":%d", port))
+}
+
+// WithSocket configures a unix socket address to connect to the .
+func WithSocket(socket string) Option {
+	withSidecar := WithSidecar(func() error {
+		return tryFixPermission(socket)
+	})
+	withSocket := With("listen", fmt.Sprintf("unix:%s", socket))
+
+	return func(o *Config) {
+		o.apply(withSidecar, withSocket)
+	}
+}
+
+// WithApiVersion sets the API version in the options with the specified version number.
+func WithApiVersion(version int) Option {
+	return With("api-version", version)
+}
+
+// Headless sets the "headless" option to true, which runs delve in server mode
+func Headless() Option {
+	return With("headless", true)
+}
+
+// WaitDebugger set the "continue" option to false,
+// the debugged application will wait until a debugger is attached, after having generating a first
+// log indicating on which endpoint it listens.
+func WaitDebugger() Option {
+	return With("continue", false)
+}
+
+// NoWait set the debugger to continue execution immediately. Debugged application
+// wouldn't wait for a debugger to be attached.
+func NoWait() Option {
+	return func(o *Config) {
+		maps.Insert(o.dlvArgs, maps.All(map[string]any{
+			"continue":           true,
+			"accept-multiclient": true,
+		}))
+	}
+}
+
+// runSidecars starts each function in the sidecars slice as a separate goroutine.
+func (o *Config) runSidecars() {
+	for _, sidecar := range o.sidecars {
+		go func() { _ = sidecar() }()
+	}
+}
+
+// tryFixPermission attempts to change the file permissions of the specified path using exponential backoff retries.
+func tryFixPermission(path string) error {
+	return retry.Call(retry.CallArgs{
+		Func: func() error { return os.Chmod(path, 0777) },
+		IsFatalError: func(err error) bool {
+			return !errors.Is(err, os.ErrNotExist)
+		},
+		Attempts:    10,
+		Delay:       200 * time.Millisecond,
+		MaxDelay:    1 * time.Second,
+		MaxDuration: 10 * time.Second,
+		BackoffFunc: func(delay time.Duration, attempt int) time.Duration {
+			return delay * time.Duration(attempt)
+		},
+		Clock: clock.WallClock,
+	})
+}
+
+// logger returns the logger implementation to use, defaulting to the predefined defaultLogger if none is specified.
+func (config *Config) logger() logger {
+	if config.definedLogger == nil {
+		return defaultLogger
+	}
+	return config.definedLogger
+}
+
+// loggerWrapper wraps a LoggerFunc to conform with the logger interface.
+type loggerWrapper struct {
+	logf LoggerFunc
+}
+
+// Printf logs a formatted message using a specified format and variadic arguments.
+func (l loggerWrapper) Printf(format string, v ...interface{}) {
+	l.logf(format, v...)
+}
+
+// defaultLogger is the default logger implementing the logger interface, using fmt.Printf to log formatted messages.
+var defaultLogger = loggerWrapper{func(format string, v ...interface{}) {
+	fmt.Printf(format, v...)
+	fmt.Println()
+}}
```

Comment:
Just use the logger package and use the Logger interface.

We don't want to create a new logger interface, since I've spent a LOT of time removing it.

Also, remove `fmt` this works well for k8s but isn't what we want to do for machines.

Reply:
The idea here is to avoid taking a decision on this package for the level of the log we want to provide, as long as letting the caller to choose how they want to pipe log of the surrounding delve server (by providing a wrapper which bind the level he want with the `Printf` implementation of the delve server.

I can remove it, anyway, and log everything as Warnf.

Reply:
In fact, that's much better with core/logger. Thanks for the insight ! (fixed btw)